api/src/test/java/org/openmrs/module/patientflags/FlagServiceTest.java

```diff
@@ -191,69 +192,72 @@ public void getFlagByUuid_shouldReturnFlagByUuid() {
 	/**
 	 *  Tests of methods that save and retrieve tags
 	 */
+	@Test
+	public void getAllTags_shouldReturnAllTags() {
+		List<Tag> tags = flagService.getAllTags();
+		assertFalse(tags.isEmpty());
+	}
+
+	@Test
+	public void getTag_shouldGetTagById() {
+		Integer id = 1;
+		Tag tag = flagService.getTag(id);
+		assertNotNull(tag);
+		assertEquals(id, tag.getTagId());
+	}
+
+	@Test
+	public void getTagByUuid_shouldGetTagByUuid() {
+		String uuid = "da7f524f-27ce-4bb2-86d6-6d1d05312bd5";
+		Tag tag = flagService.getTagByUuid(uuid);
+		assertNotNull(tag);
+		assertEquals(uuid, tag.getUuid());
+	}
+
+	@Test
+	public void getTag_shouldGetTagByName() {
+		String tagName = "high";
+		Tag tag = flagService.getTag(tagName);
+		assertNotNull(tag);
+		assertEquals(tagName, tag.getName());
+	}
+
 	@Test
 	public void addOrUpdateTag_shouldSaveNewTag() {
-		// insert a "test" tag
-		Context.getService(FlagService.class).saveTag(new Tag("test"));
+		Tag tag = createTag();
+		flagService.saveTag(tag);
 
-		// get all tags
-		List<Tag> tags = Context.getService(FlagService.class).getAllTags();
-		// since there is no test data loaded, the first (and only) tag should be the "test" one we just added
-		assertEquals("test",tags.get(0).getName());
+		Tag savedTag = flagService.getTag("allergy");

```

Comment:
Instead of using the hard-coded string 'allergy' when retrieving the tag, can we use  tag.getName() ?

Reply:
removed the hard coded values